### DISCUSSION TITLE
Deployment

### DISCUSSION THREAD

**Discussion by lsadehaan** [zzt93/syncer#17] [Ideas]
Hi, I just came across this project and it seems very useful.
So far haven't been able to see exactly the best way to deploy it though.
I'm going to check it out and then create a docker based deployment, will try to contribute it once done.

**Comment by zzt93** [comment]
Thank you for your contribution. It is my pleasure if it helps you or others.

--------------原始邮件--------------
发件人："lsadehaan ***@***.***&gt;;
发送时间：2021年3月29日(星期一) 晚上10:16
收件人："zzt93/syncer" ***@***.***&gt;;
抄送："Subscribed ***@***.***&gt;;
主题：[zzt93/syncer] Deployment (#17)
-----------------------------------

 

 
Hi, I just came across this project and it seems very useful.
 So far haven't been able to see exactly the best way to deploy it though.
 I'm going to check it out and then create a docker based deployment, will try to contribute it once done.

**Comment by lsadehaan** [comment]
Hi, I'm still doing some testing and having some difficulty with a few things.
1. How can I prevent syncer from starting from the start of the binlog every time it restarts. I get the following in my log:
Meta file in FileBasedMap{path=LocalMetaFile{path=/data/syncer/input/last_position/simple/10.0.0.7:3306}, lastRemoved=null, map={}} crashed, take as fresh run
The last_position file is there, ad contains: mysql-bin.005536/292464251/74/0
why does it not start from that point?
2. Is there an easy way to make it skip to the last binlog entry whenever it is started up? For the case where I don't want to sync from the start?

**Reply by zzt93** [reply]
A sample is [here](https://github.com/zzt93/syncer/blob/master/test/config/latest/consumer/latest.yml), the core is `autoOffsetReset: latest`
```yml
  - connection:
      clusterNodes: [${MYSQL_ADDR}]
      autoOffsetReset: latest
```

**Reply by zzt93** [reply]
The reason why it not start from that point, in my experience, is the syncer shutdown too soon or no data synced at all (one opt point to do in future) which fails to flush the position to file.

**Reply by lsadehaan** [reply]
Thanks a lot for the help!
